battle.go

```diff
@@ -259,16 +267,32 @@ func (c *battleContext) Opponents() []int {
 
 // An abstration over all possible actions an `Agent` can make in one round. Each Pokemon gets one turn.
 type Turn interface {
-	Priority() int // Gets the turn's priority. Higher values go first.
+	Priority() int     // Gets the turn's priority. Higher values go first. Not to be confused with Move priority.
+	BattleLog() string // Get a string representation of what happened in this turn. Intended to be shown to end users.
 }
 
 // Indicates that the Pokemon will fight. The active Pokemon indicated by `Self()` in the battle context will use
 // the move at `moveIdx` on the active Pokemon indicated by `targetIdx`.
 type FightTurn struct {
 	moveIdx   int // Denotes which of the pokemon's moves to use.
 	targetIdx int // The active pokemon that on the receiving end of the move.
+	Result    struct {
+		User   *Pokemon
+		Move   *Move
+		Target *Pokemon
+		Damage uint
+	}
 }
 
 func (turn FightTurn) Priority() int {
 	return 0
 }
+
+func (turn FightTurn) BattleLog() string {
+	return fmt.Sprintf("%s used %s on %s for %d damage.",
```

Comment:
I think this a good start, but I think it overgeneralizes what the result of a FightTurn might be. What about moves that heal? What about moves that change stats? How will this be handled in `BattleLog()` in the future?